src/import-data/useImportRequest.ts

```diff
@@ -0,0 +1,136 @@
+import { useRoute } from 'src/libs/nav';
+
+import {
+  CatalogDatasetImportRequest,
+  CatalogSnapshotsImportRequest,
+  FileImportRequest,
+  ImportRequest,
+  TDRSnapshotExportImportRequest,
+  TDRSnapshotReferenceImportRequest,
+} from './import-types';
+
+type QueryParams = { [key: string]: unknown };
+
+/**
+ * Validate that a value is a string. Throw an error if it is not.
+ *
+ * @param value The value to validate.
+ * @param label Label for the value to use in error messages.
+ * @returns The validated string value.
+ */
+const requireString = (value: unknown, label = 'value'): string => {
+  if (value === undefined || value === null) {
+    throw new Error(`A ${label} is required`);
+  }
+  if (typeof value !== 'string') {
+    throw new Error(`Invalid ${label}: ${value}`);
+  }
+  return value;
+};
+
+/**
+ * Get the requested import format from query parameters. Default to 'bagit' if none is specified.
+ *
+ * @param queryParams Query parameters from import request.
+ * @returns Requested import format.
+ */
+const getFormat = (queryParams: QueryParams): string => {
```

Comment:
Can this return a union type of all known formats?

Reply:
Currently, this may return any string and `getImportRequest` handles validating that it's a known/handled format.